### PR TITLE
Created Extension to submit data to OpenTSDB

### PR DESCRIPTION
**Pull request by johann8384**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/opentsdb.inc.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2014 Neil Lathwood <https://github.com/laf/ http://www.lathwood.co.uk/fa>
```
Comment (murrant) [inline includes/opentsdb.inc.php]: You should include your own copyright info here

File: includes/opentsdb.inc.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+{
+    global $opentsdb;
+
+    if ($opentsdb !== false) {
+        $timestamp = time();
+        $hostname_tag = sprintf('%s=%s ', 'hostname', $device['hostname']);
+        $tmp_tags = array($hostname_tag);
+
+        foreach ($tags as $k => $v) {
+            $v = preg_replace(array('/ /','/,/','/=/'), '_', $v);
```
Comment (murrant) [inline includes/opentsdb.inc.php]: You can just use str_replace here:
```php
$v = str_replace(array(' ',',','='), '_', $v);
```

And also move this inside the `if (!empty($v)` check

File: includes/opentsdb.inc.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            if (!empty($v)) {
+                array_push($tmp_tags, sprintf('%s=%s ', $k, $v));
+            }
+        }
+
+        foreach ($fields as $k => $v) {
+            $line = sprintf('put %s %d %d %s', $measurement, $timestamp, $v, $tmp_tags);
+            d_echo("Sending to OPenTSDB: $line\n");
+            fwrite($opentsdb, $line . "\n");
```
Comment (murrant) [inline includes/opentsdb.inc.php]: Is there any sort of indication if things fail here?  Should probably notify the user, for example if the OpenTSDB service can't be contacted.

File: includes/datastore.inc.php [hunk 4]
```diff
@@ -67,5 +67,6 @@ function data_update($device, $measurement, $tags, $fields)
 
     rrdtool_data_update($device, $measurement, $tags, $fields);
     influx_update($device, $measurement, rrd_array_filter($tags), $fields);
+    opentsdb_update($device, $measurement, rrd_array_filter($tags), $fields);
```
Comment (murrant) [inline includes/datastore.inc.php]: Please extract rrd_array_filter($tags) to a variable so it isn't called twice.

**Comment by laf** [comment]
@johann8384 Any update on these changes?

**Comment by johann8384** [comment]
I'll make them soon.

**Comment by laf** [comment]
Ace, thanks :)

**Comment by laf** [comment]
@johann8384 bump

**Comment by laf** [comment]
@johann8384 any updates?

**Comment by laf** [comment]
Feel free to let us know when you've resolved the comments in this and we can re-open.

**Pull request closed without merging**